### PR TITLE
[feat]Husky + eslint cleanup

### PR DESCRIPTION
**Pull request by RajKumar-BackCountry**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,18 +1,19 @@
 module.exports = {
   env: {
     browser: true,
+    jest: true,
     es2021: true
   },
   extends: ['airbnb-base', 'plugin:prettier/recommended'],
   overrides: [],
-  plugins: ['unused-imports', 'import'],
+  // plugins: ['unused-imports', 'import'],
```
Comment (nikhilchandrabackcountry) [inline .eslintrc.js]: add it back

File: .eslintrc.js [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
   parserOptions: {
     ecmaVersion: 'latest',
     sourceType: 'module'
   },
   rules: {
     // Indentation
-    indent: ['error', 2],
+    // indent: ['error', 2],
```
Comment (nikhilchandrabackcountry) [inline .eslintrc.js]: add it back, else code indentation will not be checked

File: .eslintrc.js [hunk 3]
```diff
@@ -49,7 +50,7 @@ module.exports = {
     'no-alert': 'error',
     'no-debugger': 'error',
     'no-unused-vars': 'error',
-    'unused-imports/no-unused-imports': 'error',
+    // 'unused-imports/no-unused-imports': 'error',
```
Comment (nikhilchandrabackcountry) [inline .eslintrc.js]: add it back

File: package.json [hunk 4]
```diff
@@ -26,6 +27,7 @@
     "express": "^4.18.2",
     "helmet": "^6.0.1",
     "joi": "^17.9.2",
+    "supertest": "^6.3.3",
```
Comment (nikhilchandrabackcountry) [inline package.json]: its a dev dependency

File: package.json [hunk 5]
```diff
@@ -36,13 +38,13 @@
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-prettier": "^4.2.1",
     "eslint-plugin-unused-imports": "^2.0.0",
+    "esquery": "^1.4.0",
```
Comment (nikhilchandrabackcountry) [inline package.json]: are we using this?

File: src/modules/groups/groups.js [hunk 6]
```diff
@@ -1,16 +1,14 @@
...
 const router = express.Router();
-// const validate = require('validate-azure-ad-token').default;
 
 const getAccessToken = async () => {
   const { CLIENT_SECRET, CLIENT_ID, DEFAULT_MS_GRAPH_SCOPE, MS_LOGIN_HOST_NAME, TENANT_ID } =
     process.env;
   try {
     const URL = `${MS_LOGIN_HOST_NAME}/${TENANT_ID}/oauth2/v2.0/token`;
-    const {
-      data: { access_token }
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: try const {
      data: { access_token: accessToken }
    }

File: src/modules/groups/groups.js [hunk 7]
```diff
@@ -24,65 +22,29 @@ const getAccessToken = async () => {
...
     return { error };
   }
 };
 
-const accessToken = async (req, res) => {
-  const { access_token, error } = (await getAccessToken()) || {};
+router.get('/token', async (req, res) => {
+  const { accessToken, error } = (await getAccessToken()) || {};
   if (error) {
     return res.status(400).send({
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: use res.sendResponse, i have created a response interceptor to have uniform response structure

File: src/modules/groups/groups.js [hunk 8]
```diff
@@ -24,65 +22,29 @@ const getAccessToken = async () => {
...
-};
+});
 
 router.get('/all/:type', async (req, res) => {
   const { type } = req.params;
-  const { MS_GRAPH_HOST_NAME, WRITERS_GROUP_ID, EDITOR_GROUP_ID } = process.env;
-  const { access_token } = (await getAccessToken()) || {};
+  const { MS_GRAPH_HOST_NAME, WRITERS_GROUP_ID, EDITOR_GROUP_ID, SIZING_GROUP_ID, ADMIN_GROUP_ID } =
+    process.env;
+  const { accessToken } = (await getAccessToken()) || {};
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: to avoid this initialization, 
update your getAccessToken function to return access token directly, remove object return

File: src/modules/groups/groups.js [hunk 9]
```diff
@@ -114,7 +76,7 @@ router.get('/all/:type', async (req, res) => {
       data: { value }
     } = await axios.get(URL, {
       headers: {
-        Authorization: `Bearer ${access_token}`
+        Authorization: `Bearer ${accessToken}`
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: good to keep Bearer concatenation in getaccesstoken method

**Review by nikhilchandrabackcountry — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 2 above)
Comment (nikhilchandrabackcountry) [reply .eslintrc.js]: you might face issue with switch case disable for that
 /* eslint-disable indent */

**Comment by nikhilchandrabackcountry** [comment]
@RajKumar-BackCountry  FYI, I have changed the "merge To" branch to have a clear idea on file changes

File: src/modules/groups/groups.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+        client_id: CLIENT_ID,
+        scope: DEFAULT_MS_GRAPH_SCOPE
+      },
+      {
+        headers: {
+          'Content-Type': 'multipart/form-data'
+        }
+      }
+    );
+    return { accessToken: data?.access_token };
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: return `Bearer ${accessToken}`

File: .eslintrc.js (hunk 1 above)
Comment (raj-kumar-litmus) [reply .eslintrc.js:9]: with this, I am not able to run eslint itself.
Getting errors.

File: src/modules/groups/groups.js (hunk 6 above)
Comment (raj-kumar-litmus) [reply src/modules/groups/groups.js]: It's not very readable. Sticking with what I have.

File: src/modules/groups/groups.js (hunk 8 above)
Comment (raj-kumar-litmus) [reply src/modules/groups/groups.js]: I am also handling an error;
So, sticking with this.

File: src/modules/groups/groups.js (hunk 9 above)
Comment (raj-kumar-litmus) [reply src/modules/groups/groups.js]: M not following this. Can you rephrase ?

File: src/modules/groups/groups.js (hunk 10 above)
Comment (raj-kumar-litmus) [reply src/modules/groups/groups.js]: What about the error scenario ?

Comment (nikhilchandrabackcountry) [reply src/modules/groups/groups.js]: return accessToken ?  `Bearer ${accessToken}` : undefined

File: package.json (hunk 4 above)
Comment (nikhilchandrabackcountry) [reply package.json]: ??

File: package.json (hunk 5 above)
Comment (nikhilchandrabackcountry) [reply package.json]: ??

File: src/modules/groups/groups.js (hunk 8 above)
Comment (nikhilchandrabackcountry) [reply src/modules/groups/groups.js]: handle in getAccessToken function

File: src/modules/groups/groups.js (hunk 9 above)
Comment (nikhilchandrabackcountry) [reply src/modules/groups/groups.js]: abstract concatenating Bearer string in getAccessToken function

File: src/modules/groups/groups.js [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+      groupId = ADMIN_GROUP_ID;
+      break;
+    case 'sizing':
+      groupId = SIZING_GROUP_ID;
+      break;
+    default:
+      groupId = null;
+  }
+
+  if (!groupId) {
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: push this logic to validate middleware 
C:\Users\nikhil.s\apps\bc-cms-rewrite-node-ms\src\middlewares.js :line 21

File: src/modules/groups/groups.js [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+  try {
+    const URL = `${MS_GRAPH_HOST_NAME}/groups/${groupId}/members`;
+    const {
+      data: { value }
+    } = await axios.get(URL, {
+      headers: {
+        Authorization: `Bearer ${accessToken}`
+      }
+    });
+    return res.status(200).send({
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: use res.sendresponse everywhere,

File: src/modules/index.js [hunk 13]
```diff
@@ -2,10 +2,12 @@ const express = require('express');
 
 const { workflows } = require('./workflows');
 const { dataNormalization } = require('./dataNormalization');
+const { groups } = require('./groups');
 
 const router = express.Router();
 
 router.use('/workflows', workflows);
 router.use('/dataNormalization', dataNormalization);
+router.use('/groupMembers', groups);
```
Comment (nikhilchandrabackcountry) [inline src/modules/index.js]: lets keep route generic as  router.use('/groups', groups);

File: src/modules/groups/groups.js (hunk 11 above)
Comment (raj-kumar-litmus) [reply src/modules/groups/groups.js]: This is not a validation logic. This is part of the api.
It should remain within this loop only.

Since there is no validation, i dont see any reason to move this to middleware.

Comment (nikhilchandrabackcountry) [reply src/modules/groups/groups.js]: i have designed this to validate query, path or body and it will take care of these validations to make sure groups can only have writers,editors, etc

with this we no need to validate path .body or query prams in request handlers

File: src/modules/groups/groups.js [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+      }
+    });
+    return res.sendResponse(
+      {
+        members: value
+      },
+      200
+    );
+  } catch (error) {
+    return res.sendResponse(
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: remove object, return res.sendResponse(
        error,
      400
    );

Comment (nikhilchandrabackcountry) [reply src/modules/groups/groups.js]: check responseInterceptor code

File: src/modules/groups/groups.js [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+  }
+  return res.sendResponse(
+    {
+      accessToken: token
+    },
+    200
+  );
+});
+
+router.get('/all/:type', async (req, res) => {
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: path should be /:type/members

File: src/modules/groups/groups.js [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+      break;
+    case 'sizing':
+      groupId = SIZING_GROUP_ID;
+      break;
+    default:
+      groupId = null;
+  }
+
+  if (!groupId) {
+    return res.sendResponse(
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: for errors send message directly return like below,

res.sendResponse(
     'groupId is required field,
      400
    );

File: src/modules/groups/groups.js [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+  );
+});
+
+router.get('/all/:type', async (req, res) => {
+  const { type } = req.params;
+  const { MS_GRAPH_HOST_NAME, WRITERS_GROUP_ID, EDITOR_GROUP_ID, SIZING_GROUP_ID, ADMIN_GROUP_ID } =
+    process.env;
+  const accessToken = (await getAccessToken()) || {};
+  if (!accessToken) {
+    return res.sendResponse({}, 400);
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: return     return res.sendResponse('unauthorized', 401);

File: src/modules/groups/groups.js [hunk 18]
```diff
@@ -0,0 +1,106 @@
+/* eslint-disable indent */
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: disable only for switch case,

File: src/modules/groups/groups.js [hunk 19]
```diff
@@ -0,0 +1,106 @@
...
+  const { type } = req.params;
+  const { MS_GRAPH_HOST_NAME, WRITERS_GROUP_ID, EDITOR_GROUP_ID, SIZING_GROUP_ID, ADMIN_GROUP_ID } =
+    process.env;
+  const accessToken = (await getAccessToken()) || {};
+  if (!accessToken) {
+    return res.sendResponse({}, 400);
+  }
+  let groupId;
+
+  switch (type) {
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js:68]: add /* eslint-disable indent */

File: src/modules/groups/groups.js (hunk 11 above)
Comment (nikhilchandrabackcountry) [reply src/modules/groups/groups.js]: for ref: C:\Users\nikhil.s\apps\bc-cms-rewrite-node-ms\src\modules\workflows\dtos.js 
once u define here, apply here 
image

Comment (RajKumar-BackCountry) [reply src/modules/groups/groups.js:72]: I dont we need to validate the incoming param.

 The switch case chooses the groupid which will be used in the subsequent Azure Api call.
I dont think we should move this logic to middleware.

File: src/modules/groups/groups.js (hunk 9 above)
Comment (nikhilchandrabackcountry) [reply src/modules/groups/groups.js]: its can return `Bearer undefined`

File: src/modules/groups/groups.js (hunk 10 above)
Comment (RajKumar-BackCountry) [reply src/modules/groups/groups.js]: THis is done.

File: src/modules/groups/groups.js (hunk 11 above)
Comment (RajKumar-BackCountry) [reply src/modules/groups/groups.js]: This, we agreed not to proceed since you werent sure about the JOI for enums.

File: src/modules/groups/groups.js (hunk 12 above)
Comment (RajKumar-BackCountry) [reply src/modules/groups/groups.js]: this is also done.

File: src/modules/index.js (hunk 13 above)
Comment (RajKumar-BackCountry) [reply src/modules/index.js]: this is done too.

File: src/modules/groups/groups.js (hunk 14 above)
Comment (RajKumar-BackCountry) [reply src/modules/groups/groups.js]: done.

File: src/modules/groups/groups.js (hunk 15 above)
Comment (RajKumar-BackCountry) [reply src/modules/groups/groups.js]: done.

File: src/modules/groups/groups.js (hunk 16 above)
Comment (RajKumar-BackCountry) [reply src/modules/groups/groups.js]: done.

File: src/modules/groups/groups.js (hunk 17 above)
Comment (RajKumar-BackCountry) [reply src/modules/groups/groups.js]: done.

File: src/modules/groups/groups.js (hunk 10 above)
Comment (nikhilchandrabackcountry) [reply src/modules/groups/groups.js]: push the changes, i don't see the changes

File: src/modules/groups/groups.js [hunk 20]
```diff
@@ -0,0 +1,87 @@
...
+    );
+    return data?.access_token ? `Bearer ${data?.access_token}` : undefined;
+  } catch (error) {
+    return null;
+  }
+};
+
+router.get('/token', async (req, res) => {
+  const token = (await getAccessToken()) || {};
+  if (!token) {
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: it will never reach this since u are initializing it with {}

File: src/modules/groups/groups.js [hunk 21]
```diff
@@ -0,0 +1,87 @@
...
+  }
+  return res.sendResponse(
+    {
+      accessToken: token
+    },
+    200
+  );
+});
+
+router.get('/all/:type/members', validateMiddleware({ params: groupDto }), async (req, res) => {
```
Comment (nikhilchandrabackcountry) [inline src/modules/groups/groups.js]: update with :type/members

File: src/modules/groups/groups.js (hunk 20 above)
Comment (RajKumar-BackCountry) [reply src/modules/groups/groups.js]: Resolved in 5cc9d909d4bb497c5c867ab54fdb7d2384701dc5
